src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -1091,11 +1091,52 @@ internal static List<DefineEnumHelper> Sort(List<DefineEnumHelper> defineEnumHel
 
             internal void DefineEnum()
             {
+                var typeConstraintAst = _enumDefinitionAst.BaseTypes.FirstOrDefault();
+                var underlyingType = typeConstraintAst == null ? typeof(int) : typeConstraintAst.TypeName.GetReflectionType();
+
                 var definedEnumerators = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-                var enumBuilder = _moduleBuilder.DefineEnum(_typeName, Reflection.TypeAttributes.Public, typeof(int));
+                var enumBuilder = _moduleBuilder.DefineEnum(_typeName, Reflection.TypeAttributes.Public, underlyingType);
                 DefineCustomAttributes(enumBuilder, _enumDefinitionAst.Attributes, _parser, AttributeTargets.Enum);
-                int value = 0;
+
+                dynamic value = 0;
+                dynamic maxValue = 0;
+                switch (Type.GetTypeCode(underlyingType))
```

Comment:
Uh, it used to do something different for enums I think, you'll have to use your favorite decompiler on an older version of PowerShell to double check though.

Reply:
In this case that isn't so interesting because we're always passing in a `Type` instance (for which every other instance I can find `Type.GetTypeCode()` is also used)

Reply:
I thought it is for future enhancement but if it was removed we can ignore this.

@IISResetMe Please don't set "Resolve conversation" until we finish the code review - it is difficult to understand that comments needs an attention. Please leave this to comment's author or maintainers.

Reply:
@iSazonov got it, sorry about that